Make Yvonne a Maintainer of PyGMT

yvonnefroehlich opened the pull request:
**Description of proposed changes**

This PR aims to update the Teams Gallery reagarding Yvonne becoming a Maintainer of PyGMT.
For context see the discussion up on https://github.com/GenericMappingTools/pygmt/pull/2651#discussion_r1312481432.

**Preview**: https://pygmt-dev--2655.org.readthedocs.build/en/2655/team.html

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

seisman commented:
@yvonnefroehlich Following the access checklist in PR https://github.com/GenericMappingTools/pygmt/pull/2656, I've added you to the [pygmt-maintainers](https://github.com/orgs/GenericMappingTools/teams/pygmt-maintainers) team, and also removed you from the [pygmt-contributors](https://github.com/orgs/GenericMappingTools/teams/pygmt-contributors) team. Please note that pygmt-maintainers is a child team of pygmt-contributors, so I presume you still will receive notification when some ping the `@GenericMappingTools/pygmt-contributors` team.

yvonnefroehlich commented:
> @yvonnefroehlich Following the access checklist in PR #2656, I've added you to the [pygmt-maintainers](https://github.com/orgs/GenericMappingTools/teams/pygmt-maintainers) team, and also removed you from the [pygmt-contributors](https://github.com/orgs/GenericMappingTools/teams/pygmt-contributors) team.

Sound good - thanks @seisman 🙂!

> Please note that pygmt-maintainers is a child team of pygmt-contributors, so I presume you still will receive notification when some ping the `@GenericMappingTools/pygmt-contributors` team.

No worries, that's no problem for me at all.
